### ISSUE TITLE
uninitialized constant JSON::PRETTY_STATE_PROTOTYPE (NameError)

### ISSUE THREAD

**Issue by thiagotrentin** [flori/json#458]
I'm getting this error when try to run my Cucumber tests that uses Json as dependency after upgrating to version 2.5.

```
uninitialized constant JSON::PRETTY_STATE_PROTOTYPE (NameError)
/usr/local/bundle/gems/multi_json-1.15.0/lib/multi_json/adapters/json_common.rb:18:in `dump'
/usr/local/bundle/gems/multi_json-1.15.0/lib/multi_json/adapter.rb:25:in `dump'
/usr/local/bundle/gems/multi_json-1.15.0/lib/multi_json.rb:139:in `dump'
/usr/local/bundle/gems/cucumber-3.2.0/lib/cucumber/formatter/json.rb:77:in `on_test_run_finished'
/usr/local/bundle/gems/cucumber-core-3.2.1/lib/cucumber/core/event_bus.rb:34:in `block in broadcast'
/usr/local/bundle/gems/cucumber-core-3.2.1/lib/cucumber/core/event_bus.rb:34:in `each'
/usr/local/bundle/gems/cucumber-core-3.2.1/lib/cucumber/core/event_bus.rb:34:in `broadcast'
/usr/local/bundle/gems/cucumber-core-3.2.1/lib/cucumber/core/event_bus.rb:40:in `method_missing'
/usr/local/bundle/gems/cucumber-3.2.0/lib/cucumber/configuration.rb:33:in `notify'
/usr/local/bundle/gems/cucumber-3.2.0/lib/cucumber/runtime.rb:76:in `run!'
/usr/local/bundle/gems/cucumber-3.2.0/lib/cucumber/cli/main.rb:34:in `execute!'
/usr/local/bundle/gems/cucumber-3.2.0/bin/cucumber:9:in `<top (required)>'
/usr/local/bundle/bin/cucumber:23:in `load'
/usr/local/bundle/bin/cucumber:23:in `<main>'
```

Could you help me with this issue?

**Comment by hsbt** [comment]
Thanks for your report.

Can you try the following patch with your local env?

```
diff --git lib/json/common.rb lib/json/common.rb
index 3e390f2d6b..ea46896fcc 100644
--- lib/json/common.rb
+++ lib/json/common.rb
@@ -72,6 +72,8 @@ def generator=(generator) # :nodoc:
       self.state = generator::State
       const_set :State, self.state
       const_set :SAFE_STATE_PROTOTYPE, State.new # for JRuby
+      const_set :FAST_STATE_PROTOTYPE, create_fast_state
+      const_set :PRETTY_STATE_PROTOTYPE, create_pretty_state
     ensure
       $VERBOSE = old
     end
```

**Comment by thiagotrentin** [comment]
> Can you try the following patch with your local env?

it works! 👏 

Thanks for your help!

**Comment by hsbt** [comment]
Great. I will release 2.5.1 with the above patch.

**Comment by hsbt** [comment]
Just released json-2.5.1.
